### ISSUE TITLE
Question: How do we handle migrations?

### ISSUE THREAD

**Issue by 0x7171** [flashfish0x/yearn-affiliate-tracker#8]
> 0xJiji, [15 Dec 2021, 16:56:02]:
> I think the key difference here is that fish’s proposal is one contract, maintained by us, for all affiliates
> 
> whereas baseRouter, or our previous thinking, has been one router per affiliate, maintained by the affiliate
> 
> Also, sth to keep in mind, I think neither approach handles vault migrations correctly right?
> 
> Say $100m has been deposited via Affiliate A into the yvDai vault, and now capital is migrated to a new version. 
> 
> Users could migrate their vault tokens to the new version of the yvDAI vault, but we wouldn’t be attributing that TVL to Affiliate A any more, right?

Seems very tricky to handle this the right way, but placing here in case it stimulates some ideas.

**Comment by flashfish0x** [comment]
Hopefully we are pretty much finished with migrations for the foreseeable future. But if we do migrate i think it is fine for partners to lose their referral given it requires the user to withdraw and redeposit to do a migrate. 

We could potentially add a migrate option to this contract but i don't think it is worth the hassle

**Comment by 0x7171** [comment]
> given it requires the user to withdraw and redeposit to do a migrate

Yes, but can they do that via the partner and still get the credit at the same time? I suppose so right, as long as partner has the new vault in their interface?
